### PR TITLE
Include Black code formatting check on Travis-CI

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR formats all of the cg code according to the Black formatting scheme

**How to test CI**:
- [x] open the travis-PR-build

**Expected test outcome**:
- [x] `cg` black section should be red.

- [ ] Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by @barrystokman @moonso 
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This is |minor| **version bump** because we change the development functionality but not the end user functionality

### CODE REVIEW

**Comment by ingkebil** [comment]
I wish you wouldn't have run `black` on the whole repo already. It makes it really hard to see what changes you actually made.

**Comment by patrikgrenfeldt** [comment]
It has four distinct commits so there should be no confusion about what my actual changes were, in what way is it hard to follow the changes?

**Comment by patrikgrenfeldt** [comment]
If one uses GitHub.com one can open the Tab "Commits" on this pull-request and view the commits one-by-one. The changes that are my changes are in the first three commits in chronological order, the fourth is the formatting that Black does and the fifth is a merge with changes made in master. 

@moonso, @henrikstranneheim we have talked a bit about this topic last week and maybe you able to follow my changes and can review them?

**Comment by patrikgrenfeldt** [comment]
Test result CI-Black:
bild
👍 
(The linting section on Travis is red due to lines formatted by Black that contains lint issues, fixing those is beyond the scope of this PR)

File: .travis.yml [hunk 1]
```diff
@@ -24,9 +24,13 @@ jobs:
         - pip install -U -r requirements-dev.txt .
         - coverage run --branch --omit="*.html" --source cg setup.py test
         - coveralls
+    - name: "code formatting"
```
Comment (moonso) [inline .travis.yml]: 👍

File: .gitlint.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ pylint:
     - .py
   command: pylint
   arguments:
-    - --rcfile={DEFAULT_CONFIGS}/pylintrc
+    - --rcfile=.configs/pylintrc
```
Comment (moonso) [inline .gitlint.yaml]: You stole this :)

Comment (patrikgrenfeldt) [reply .gitlint.yaml]: Well it's the fix you did in the Subprocess PR :)

**Comment by patrikgrenfeldt** [comment]
Thanks @moonso for taking time to do the review!

**Comment by moonso** [comment]
I think you should merge this

**Comment by patrikgrenfeldt** [comment]
Closing PR since it became to messy with the reverts, the few actual changes are made in #499
